### PR TITLE
CSA-51

### PR DESCRIPTION
**Pull request by VovaLyapota**

### CODE REVIEW

**Review by tymnik — COMMENTED** [review]
Now you need to add padding-bottom at login page

**Review by tymnik — APPROVED** [review]

File: src/components/AvatarInput/AvatarInput.tsx [hunk 1]
```diff
@@ -1,35 +1,52 @@
...
-        />
+      <input
+        {...register}
+        id="userAvatar"
+        className="sr-only"
+        type="file"
+        accept="image/*"
+        onChange={(e) => {
+          const file = e.target.files && e.target.files[0];
+          if (file) {
```
Comment (StanislavKharkiv) [inline src/components/AvatarInput/AvatarInput.tsx]: A little refactor: you can write with optional chaining  `e.target.files?.[0]`

File: src/components/EditProfileForm/EditProfileForm.tsx [hunk 2]
```diff
@@ -1,10 +1,122 @@
...
+  selectUser,
+} from '../../store/user/selectors';
+import { deleteUser, updateUser } from '../../store/user/thunks';
+import AvatarInput from '../AvatarInput/AvatarInput';
+import Button from '../Button';
+import Icon from '../Icon';
+import Input from '../Input';
+import Spinner from '../Spinner';
 import styles from './EditProfileForm.module.css';
+import { createFormData } from '../../utils';
```
Comment (StanislavKharkiv) [inline src/components/EditProfileForm/EditProfileForm.tsx]: Please put styles in the end like in other components

File: src/components/UserNav/UserNav.module.css [hunk 3]
```diff
@@ -1,42 +1,77 @@
...
+  width: 100%;
+  height: 100%;
+}
+
+.dropdownMenu {
+  position: absolute;
+  top: 68px;
+  left: 0;
+  /* z-index: 2; */
+
```
Comment (StanislavKharkiv) [inline src/components/UserNav/UserNav.module.css]: why z-index in comment? you don't need it? can we clear it?

File: src/components/UserNav/UserNav.tsx [hunk 4]
```diff
@@ -1,34 +1,120 @@
...
+
+  const toggleDropdown = () => setShouldShowDropdown((p) => !p);
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (
+      dropdownRef.current &&
+      !dropdownRef.current.contains(event.target as Node)
+    ) {
+      setShouldShowDropdown(false);
+    }
```
Comment (StanislavKharkiv) [inline src/components/UserNav/UserNav.tsx]: The same, try `dropdownRef.current?.contains(event.target as Node)`

File: src/components/UserNav/UserNav.tsx [hunk 5]
```diff
@@ -1,34 +1,120 @@
...
+    if (shouldShowDropdown) {
+      window.addEventListener('click', handleClickOutside);
+    } else {
+      window.removeEventListener('click', handleClickOutside);
+    }
+
+    return () => {
+      window.removeEventListener('click', handleClickOutside);
+    };
+  }, [shouldShowDropdown]);
```
Comment (StanislavKharkiv) [inline src/components/UserNav/UserNav.tsx]: could you just once add listener and remove when component destroy?

File: src/schemas/registerSchema.ts [hunk 6]
```diff
@@ -16,7 +15,7 @@ export const registerSchema = yup.object().shape({
     .string()
     .oneOf([yup.ref('password')], MESSAGES.PASSWORD_REPEAT)
     .required(MESSAGES.PASSWORD_REPEAT_REQUIRED),
-  avatar: yup.mixed().notRequired(),
+  avatar: yup.mixed(),
   // .test('fileSize', 'The file is too large', (value) => {
```
Comment (StanislavKharkiv) [inline src/schemas/registerSchema.ts:19]: Do you need this part?

File: src/store/user/thunks.ts [hunk 7]
```diff
@@ -34,12 +45,66 @@ export const login = createAsyncThunk<
...
+export const updateUser = createAsyncThunk<
+  Pick<UserState, 'user'>,
+  { id: UserId; credentials: EditProfilePropsType },
+  {
+    rejectValue: string;
+  }
+>('user/update', async ({ id, credentials }, thunkAPI) => {
+  try {
+    console.log(id);
+    const user = await updateUserById(id, credentials);
```
Comment (StanislavKharkiv) [inline src/store/user/thunks.ts]: what console.log do here?

File: src/components/EditProfileForm/EditProfileForm.tsx [hunk 8]
```diff
@@ -1,10 +1,122 @@
...
+        <Button
+          type="button"
+          variant="link"
+          icon={<Icon id="trash" />}
+          iconPosition="before"
+          onClick={async () => {
+            const { payload } = await dispatch(deleteUser(id));
+            payload === true && navigate(ROUTES.REGISTER);
+          }}
+          className={styles.trashButton}
```
Comment (StanislavKharkiv) [inline src/components/EditProfileForm/EditProfileForm.tsx]: Better write function for onClick cb if they more than one line long. I makes our jsx layout more clear

File: src/components/UserNav/UserNav.tsx [hunk 9]
```diff
@@ -1,34 +1,116 @@
...
+        })}
+        <span className={styles.separator} />
+        <Link
+          to={ROUTES.LOGIN}
+          className={styles.dropdownItem}
+          onClick={() => {
+            deleteToken();
+            dispatch(logout());
+            toggleDropdown();
+          }}
```
Comment (StanislavKharkiv) [inline src/components/UserNav/UserNav.tsx]: The same, better make function and pass here for clear our layout

**Review by StanislavKharkiv — APPROVED** [review]

**Pull request merged**